### PR TITLE
Vendor: update library-go for ibmcloud provider

### PR DESCRIPTION
**Pull request by jonesbr17**
**- What I did**
Revendor dependencies

**- How to verify it**

**- Description for the changelog**
Pull in support for ibmcloud added in https://github.com/openshift/library-go/pull/1161

### CODE REVIEW

**Comment by sinnykumari** [comment]
This PR is also updating some of the kubernetes deps to 1.22.1. Can you please also update rest of the kube deps to 1.22.1 as well?
Also, since MCO uses replace directive to pin some kube and OpenShift deps, you will need to version match deps updated in require section into replace (https://github.com/openshift/machine-config-operator/blob/master/go.mod#L64) section, otherwise vendored content doesn't get updated with correct versions.

**Review by rtheis — COMMENTED** [review]
Thanks Bryan.  No further comments beyond what was already provided.

File: go.mod [hunk 1]
```diff
@@ -35,9 +35,9 @@ require (
 	github.com/imdario/mergo v0.3.12
 	github.com/mattn/go-isatty v0.0.12 // indirect
 	github.com/opencontainers/go-digest v1.0.0
-	github.com/openshift/api v0.0.0-20210817132244-67c28690af52
-	github.com/openshift/client-go v0.0.0-20210730113412-1811c1b3fc0e
-	github.com/openshift/library-go v0.0.0-20210819104210-e14e06ba8d47
+	github.com/openshift/api v0.0.0-20210831091943-07e756545ac1
```
Comment (Prashanth684) [inline go.mod:38]: a request from me would also be to update `openshift/api` to the latest so we can pick up the changes for the PowerVS platform which will be supported soon(which is this commit: https://github.com/openshift/api/commit/3611f00b94fd7828c1e94f5c54997bc085a74676)

File: go.mod [hunk 2]
```diff
@@ -73,30 +73,34 @@ replace (
...
-	k8s.io/kubelet => k8s.io/kubelet v0.22.0-rc.0
-	k8s.io/kubernetes => k8s.io/kubernetes v1.22.0-rc.0
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.22.0-rc.0
-	k8s.io/metrics => k8s.io/metrics v0.22.0-rc.0
-	k8s.io/mount-utils => k8s.io/mount-utils v0.23.0-alpha.0
-	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.22.0-rc.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.22.0-rc.0
+	k8s.io/api => k8s.io/api v0.22.1
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.22.1
+	k8s.io/apimachinery => k8s.io/apimachinery v0.23.0-alpha.0
```
Comment (rtheis) [inline go.mod:78]: Why aren't these all consistent?

File: go.mod [hunk 3]
```diff
@@ -73,30 +73,34 @@ replace (
...
+	k8s.io/kubelet => k8s.io/kubelet v0.22.1
+	k8s.io/kubernetes => k8s.io/kubernetes v1.22.1
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.22.1
+	k8s.io/metrics => k8s.io/metrics v0.22.1
+	k8s.io/mount-utils => k8s.io/mount-utils v0.22.2-rc.0
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.22.1
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.22.1
 )
+
+replace k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.22.1
```
Comment (sinnykumari) [inline go.mod:104]: why are these additional replace got added?

File: go.mod [hunk 4]
```diff
@@ -73,30 +73,34 @@ replace (
...
+	k8s.io/apiserver => k8s.io/apiserver v0.22.1
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.1
+	k8s.io/client-go => k8s.io/client-go v0.22.1
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.22.1
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.22.1
+	k8s.io/code-generator => k8s.io/code-generator v0.22.2-rc.0
+	k8s.io/component-base => k8s.io/component-base v0.22.1
+	k8s.io/component-helpers => k8s.io/component-helpers v0.22.1
+	k8s.io/controller-manager => k8s.io/controller-manager v0.22.1
+	k8s.io/cri-api => k8s.io/cri-api v0.23.0-alpha.0
```
Comment (sinnykumari) [inline go.mod:88]: should be consistent with rest of the kube version i.e. v0.22.1

File: go.mod [hunk 5]
```diff
@@ -73,30 +73,34 @@ replace (
...
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.22.0-rc.0
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.22.0-rc.0
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.22.0-rc.0
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.22.0-rc.0
-	k8s.io/kubectl => k8s.io/kubectl v0.22.0-rc.0
-	k8s.io/kubelet => k8s.io/kubelet v0.22.0-rc.0
-	k8s.io/kubernetes => k8s.io/kubernetes v1.22.0-rc.0
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.22.0-rc.0
-	k8s.io/metrics => k8s.io/metrics v0.22.0-rc.0
-	k8s.io/mount-utils => k8s.io/mount-utils v0.23.0-alpha.0
```
Comment (sinnykumari) [inline go.mod:99]: @umohnani8 Do we need to keep this version or we can keep it consistent with rest of the replace for kube deps i.e v0.22.1

File: go.mod [hunk 6]
```diff
@@ -73,30 +73,34 @@ replace (
 	github.com/openshift/api => github.com/openshift/api v0.0.0-20210817132244-67c28690af52
```
Comment (sinnykumari) [inline go.mod:73]: You need to update openshift/api version as well to v0.0.0-20210831091943-07e756545ac1

**Comment by sinnykumari** [comment]
Latest change mostly looks good. One question:
@jonesbr17 did you run `make go-deps` (https://github.com/openshift/machine-config-operator/blob/master/Makefile#L54)  after updating the go.mod file. Asking this because I don't see much changes in vendor directory with updated content.

File: go.mod (hunk 6 above)
Comment (clnperez) [reply go.mod]: the installer is bumping to v0.0.0-20210907212721-c9d6fcc62893 -- https://github.com/openshift/installer/pull/5198/files, which would pick up the api commits that @Prashanth684 mentioned

File: go.mod [hunk 7]
```diff
@@ -70,33 +70,33 @@ replace (
...
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.22.0-rc.0
+	k8s.io/api => k8s.io/api v0.22.1
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.22.1
+	k8s.io/apimachinery => k8s.io/apimachinery v0.22.1
+	k8s.io/apiserver => k8s.io/apiserver v0.22.1
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.1
+	k8s.io/client-go => k8s.io/client-go v0.22.1
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.22.1
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.22.1
+	k8s.io/code-generator => k8s.io/code-generator v0.22.2-rc.0
```
Comment (rtheis) [inline go.mod:84]: Why isn't this `v0.22.1`?

File: go.mod [hunk 8]
```diff
@@ -70,33 +70,33 @@ replace (
...
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.22.1
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.22.1
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.22.1
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.22.1
+	k8s.io/kubectl => k8s.io/kubectl v0.22.1
+	k8s.io/kubelet => k8s.io/kubelet v0.22.1
+	k8s.io/kubernetes => k8s.io/kubernetes v1.22.1
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.22.1
+	k8s.io/metrics => k8s.io/metrics v0.22.1
+	k8s.io/mount-utils => k8s.io/mount-utils v0.22.2-rc.0
```
Comment (rtheis) [inline go.mod:99]: Why isn't this `v0.22.1`?

**Comment by sinnykumari** [comment]
Please fix some final comments from @rtheis and then this should be good to get merged

**Comment by jeffnowicki** [comment]
@sinnykumari can you add required label(s)... `lgtm` and/or `approved`? would like to get this merged and into release image. thank you

**Comment by sinnykumari** [comment]
> @sinnykumari can you add required label(s)... `lgtm` and/or `approved`? would like to get this merged and into release image. thank you

There are few outstanding comment that needs to get fixed merge https://github.com/openshift/machine-config-operator/pull/2757#discussion_r705659660 and https://github.com/openshift/machine-config-operator/pull/2757#discussion_r705659592

File: go.mod (hunk 7 above)
Comment (jonesbr17) [reply go.mod]: I believe they point to the same code level, but I updated for clarity

**Comment by JoelSpeed** [comment]
/lgtm

Thanks @jonesbr17

**Comment by sinnykumari** [comment]
Skipping optional tests to save some ci resources.
/skip

**Review by rtheis — APPROVED** [review]
/lgtm

**Pull request merged**